internal/util/util.go

```diff
@@ -423,3 +428,27 @@ func Sample(m proto.Message, strategy *SamplingStrategy) proto.Message {
 	})
 	return m
 }
+
+// HashStore is a simple a store to produce hash for protobuf message
+type HashStore struct {
+	store map[string]int
+}
+
+// GetHash retrieves a hash string for a given protobuf message.
+// Note this should be restrict to a request scope.
```

Comment:
Does this mean across different requests, for the same proto message, we could get different hashes?

Reply:
Changed to use hash function that generates deterministic number